include/edm4hep/GenToolInfo.h

```diff
@@ -0,0 +1,47 @@
+#ifndef EDM4HEP_GENTOOLINFO_H
+#define EDM4HEP_GENTOOLINFO_H
+
+#include "podio/Frame.h"
+#include "Constants.h"
+#include <string>
+#include <vector>
+
+namespace edm4hep {
+
+struct GenToolInfo {
+  std::string name;
+  std::string version;
+  std::string description;
+};
+
+const std::vector<GenToolInfo> getGenToolInfos(podio::Frame& frame) {
+  auto toolInfos = std::vector<GenToolInfo>();
+  auto names = frame.getParameter<std::string>("name");
+  auto versions = frame.getParameter<std::string>("version");
+  auto descriptions = frame.getParameter<std::string>("description");
```

Comment:
I think the parameter names / keys need to a bit more descriptive here. Otherwise the chances for clashes is quite high. We should also put them into our `Constants.h` header so that there is a sort of canonical form to refer to them.

Reply:
yap. that's a good point. I was considering putting that into the constants indeed. Next iteration will have it